main.cpp

```diff
@@ -232,6 +271,29 @@ struct IntType
         IntType& pow( const DoubleType& operand );
         IntType& pow( const IntType& operand );
         IntType& pow( int operand );
+
+        IntType& apply(std::function<IntType&(int&)> callable)
+        {
+            if(callable)
+            {
+                return callable(*heapInt); 
+            }
+            else
+            {
+                return *this;
+            }
```

Comment:
You have no code that runs after the `else` block.  Therefore you do not need this `else` keyword
- [ ] revise